Content.Server/Teleportation/Components/DimensionPotComponent.cs

```diff
@@ -0,0 +1,62 @@
+using Content.Server.Teleportation.Systems;
+using Robust.Shared.Audio;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Teleportation.Components;
+
+/// <summary>
+/// Creates a pocket dimension map on spawn.
+/// When activated by alt verb, spawns a portal to this dimension or closes it.
+/// </summary>
+[RegisterComponent]
+[Access(typeof(DimensionPotSystem))]
+public sealed class DimensionPotComponent : Component
+{
+    /// <summary>
+    /// The portal on the pot, if it is open right now.
+    /// </summary>
+    [ViewVariables]
+    public EntityUid? PotPortal = null;
+
+    /// <summary>
+    /// The portal in the pocket dimension, usually exists.
+    /// </summary>
+    [ViewVariables]
+    public EntityUid? DimensionPortal = null;
+
+    /// <summary>
+    /// Map of the pocket dimension, usually exists.
+    /// </summary>
+    [ViewVariables]
+    public MapId PocketDimensionMap = MapId.Nullspace;
```

Comment:
Needs datafields and pocketdimensionmap should probably be an entityuid.

Reply:
also how do you delete a map using its entityuid

Reply:
No?